File: Greedy/45-jump-game-ii/jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int n = nums.length - 1;
        int l = 0;
        int r = 0;
        int jumps = 0;

        while (r < n) {
            int jr = r;
            for (int i = l; i <= r; i++) {
                jr = Math.max(jr, i + nums[i]);
            }
            l = r + 1;
            r = jr;
            jumps++;
        }

        return jumps;
    }
}

File: Greedy/45-jump-game-ii/jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums) - 1
        l = 0
        r = 0
        jump = 0

        while r < n:
            jr = r
            for i in range(l, r + 1):
                jr = max(jr, i + nums[i])
            l = r + 1
            r = jr
            jump += 1
        
        return jump